kivy/uix/widget.py

```diff
@@ -1477,3 +1547,5 @@ def dec_disabled(self, count=1):
         :class:`~kivy.properties.AliasProperty` to allow access to its
         previous state when a parent's disabled state is changed.
     '''
+
+    motion_filter = DictProperty()
```

Comment:
Added the docstings 3d8717e0ebb0f02f412a21ce31426fcc0b8c0744, but we need a larger documentation to explain the entire system as `motion_filter` is one part of it. Maybe `kivy/eventmanager/__init__.py` is a good place and then other parts can reference it.

Interesting feature of motion event `type_id` is that it can be set (stored/restored) by event manager if user wants a finer filtering of events. For example if user want to handle only touch events from the mouse (only instances of `MouseMotionEvent`) then:

- With current system user would have to override `on_touch_xxx` methods and do the `isinstance` test. Widgets will receive both `MouseMotionEvent` instances along with the all other types.
- With event manager, user can define a manager, do the `isinstance` test in the manager, set `me.type_id` to "mouse_touch", dispatch events through event tree and widgets who registered for "mouse_touch" (called `self.register_for_motion_event('mouse_touch')`) will receive those events. This way manager is doing the filtering instead of every widget.

Reply:
Looks like a much more efficient way to handle touch events. But I'm not clear on the interactions with existing Widget's on_touch_xxx() behavior.

For example if a `ScreenManager` `register_for_motion_event('mouse_touch')` ,are the on_motion events filtered during a screen transition? 
https://github.com/kivy/kivy/blob/master/kivy/uix/screenmanager.py#L1202-L1215 

The background to my question is I have been working on a gestures class https://github.com/Android-for-Python/gestures4kivy and it would seem natural for it to register it's self and override on_motion rather than on_touch_xxx (If I understand the comments above). But in the `ScreenManager` case that would require on_motion events be filtered during a screen transition like on_touch_xxx . This is the only case I know of like this, but there may be others.

Reply:
If any of the registered event manager do not handle received touch event, then that event will be dispatched through `on_touch_xxx` events (it fallbacks to the current system). Take a look at the code in `Window.on_motion` of this pull request.

I'll add:
```
def on_motion(self, etype, me):
    if self.transition.is_active:
        return False
    return super(ScreenManager, self).on_motion(etype, me)
```
to `ScreenManager` and that will stop the manager from handling the events any further. Thanks for pointing this out.

For example, you can create event manager to handle "two fingers move" gesture which you can generate from touch events. Manager would have to declare to receive "touch" events (motion events with `type_id` set to "touch"), detect the gesture, create new event (with its `type_id` set to "two_fingers_move") to hold data about the gesture and dispatch it to the widgets. Widgets registered to receive "two_fingers_move" (with `register_for_motion_event('two_motion_move')` will receive that event in `on_motion` method.

Reply:
Thanks, "If any of the registered event manager do not handle received ...." explained it.